android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -710,6 +719,13 @@ private void disconnect(Result poResult) {
                 final ConnectivityManager connectivityManager = (ConnectivityManager) moContext.getSystemService(Context.CONNECTIVITY_SERVICE);
                 connectivityManager.unregisterNetworkCallback(networkCallback);
             }
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
+                if (networkSuggestions != null) {
+                    moWiFi.removeNetworkSuggestions(networkSuggestions);
+                }
+            } else {
+                moWiFi.removeNetworkSuggestions(moWiFi.getNetworkSuggestions());
```

Comment:
Just a small comment on this: as far as I see `networkSuggestions` is always populated, recreated at every request (so we should be careful if one tries to call `connect` again before a `disconnect`), we should always have it with the correct values. 
Since `getNetworkSuggestions()` is indeed API >= 30 we cannot use it always, but at that point if we want to keep backwards compatibility I would remove the `else` branch to keep code cleaner, so just always:
```
            if (networkSuggestions != null) {
                moWiFi.removeNetworkSuggestions(networkSuggestions);
             }
```
This works for both pre and post level 30 and ensures that the exact same behaviour is kept (using the `getNetworkSuggestions()` may return more than just the last if we populate it otherwise at some point for some reason).

Reply:
I agree with this suggestion, implementing it.

Reply:
Implemented in d41bcb22d1a2cc2240dd573b41729150dca70b29